### PR TITLE
87 fix plugin review issues 2

### PR DESCRIPTION
**Pull request by benlk**
This is a placeholder PR for keeping track of the net diff between these branches as we work.

## Why

For #87: sanitizing, validating, and escaping use of $_POST.

### CODE REVIEW

**Comment by jonathanstegall** [comment]
I'm adding us both as reviewers so we can easily see what changes are

**Comment by benlk** [comment]
Before merging this, we'll need to recompile the pot and the docs, right?

**Comment by jonathanstegall** [comment]
@benlk yes. Running `gulp` recompiles the pot, and `apigen generate` makes the docs again. I'll do that now.

File: classes/salesforce_mapping.php [hunk 1]
```diff
@@ -121,38 +125,39 @@ public function create_fieldmap( $posted = array(), $wordpress_fields = array(),
...
 					if ( 'salesforce_record_type' === $key ) {
-						$record_type = $value;
+						$record_type = sanitize_text_field( $value );
 					} else {
 						$i++;
 						if ( $i > 1 ) {
 							$where .= ' AND ';
 						}
-						$where .= '`' . $key . '`' . ' = "' . $value . '"';
+						$where .= '`' . $key . '` = "' . $value . '"';
```
Comment (jonathanstegall) [inline classes/salesforce_mapping.php]: I think I'm confused by this line. I don't know how to tell if it is doing the same thing it was doing, or if what it was doing had a problem?

Comment (benlk) [inline classes/salesforce_mapping.php]: it goes from using PHP to concatenate string ```'`'``` to string ```' = "'```
to just doing that in the file and saving us a `.`

This was something that https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards brought up.

**Comment by jonathanstegall** [comment]
This PR does a few things:

1. Sanitize, validate, or escape data before saving or outputting it.
2. Fix WordPress code formatting style issues.
3. Allow output strings to be translated (and escape them after they are translated). Also updated the pot file for languages.
4. Fix bugs related to salesforce_pull delete and upsert.

**Comment by benlk** [comment]
@jonathanstegall Would you like to go ahead and merge this?

**Comment by jonathanstegall** [comment]
@benlk I just didn't want to be the only one to review. I put RC on to review as well, but if you have already reviewed that is fine with me. We can let him review too, but can go ahead and merge. Just a lot of changes.

**Pull request merged**